src/clojure/nrepl/misc.clj

```diff
@@ -89,3 +90,28 @@
   []
   (.startsWith (System/getProperty "java.runtime.version")
                "1.8"))
+
+(def var-meta-whitelist
+  "A list of var metadata attributes are safe to return to the clients.
+  We need to guard ourselves against EDN data that's not encodeable/decodable
+  with bencode. We also optimize the response payloads by not returning
+  redundant metadata."
+  [:ns :name :doc :file :arglists :forms :macro :special-form
+   :protocol :line :column :added :deprecated :resource])
+
+(defn resolve-file
```

Comment:
We should probably add a dostring and a test for this one as well.

Reply:
Good point, will do.

Reply:
In fact, since `normalize-meta`/`sanitise-meta` is the only call site, I think we can just inline this function instead. As in:

```clojure
(update :file #(or (some-> % io/resource str) %))
```

After all, the only thing the test for `normalize-meta` currently tests for is actually the behavior of the `resolve-file` function.